ctapipe/reco/hillas.py

```diff
@@ -71,44 +82,85 @@ def hillas_parameters_1(pix_x, pix_y, image):
     image = np.asanyarray(image, dtype=np.float64)
     assert pix_x.shape == image.shape
     assert pix_y.shape == image.shape
-
+    
     # Compute image moments
-    _s = np.sum(image)
-    m_x = np.sum(image * pix_x) / _s
-    m_y = np.sum(image * pix_y) / _s
-    m_xx = np.sum(image * pix_x * pix_x) / _s  # note: typo in paper
-    m_yy = np.sum(image * pix_y * pix_y) / _s
-    m_xy = np.sum(image * pix_x * pix_y) / _s  # note: typo in paper
-
-    # Compute major axis line representation y = a * x + b
-    S_xx = m_xx - m_x * m_x
-    S_yy = m_yy - m_y * m_y
-    S_xy = m_xy - m_x * m_y
-    d = S_yy - S_xx
-    temp = d * d + 4 * S_xy * S_xy
-    a = (d + np.sqrt(temp)) / (2 * S_xy)
-    b = m_y - a * m_x
+    size = np.sum(image)
+
+    #Sanity check1:
+    if size == 0:
+       print ("Empty pixels!")
+       sys.exit(1)
+
+    mean_x = np.sum(image * pix_x) / size
+    mean_y = np.sum(image * pix_y) / size
+
+    
+    # Compute major axis line representation y = a * x + b and correlations
+    S_xx = np.sum(image * (pix_x - mean_x) ** 2 ) / size
+    S_yy = np.sum(image * (pix_y - mean_y) ** 2 ) / size
+    S_xy = np.sum(image * (pix_x - mean_x) * (pix_y - mean_y) ) / size
+    S_xxx = np.sum(image * (pix_x - mean_x) ** 3 ) / size
+    S_yyy = np.sum(image * (pix_y - mean_y) ** 3 ) / size
+    S_xyy = np.sum(image * (pix_x - mean_x) * (pix_y - mean_y) ** 2 ) / size
+    S_xxy = np.sum(image * (pix_y - mean_y) * (pix_x - mean_x) ** 2 ) / size
+    S_x4 = np.sum(image * (pix_x - mean_x) ** 4 ) / size
+    S_y4 = np.sum(image * (pix_y - mean_y) ** 4 ) / size
+
+    #Sanity check2:
+
+    #If S_xy=0 (which should happen not very often, because Size>0) we cannot calculate Length and Width.
+    #In reallity it is almost impossible to have a distribution of cerenkov photons in the used pixels which is exactly symmetric
+    # along one of the axis.
+    if S_xy == 0:
+       print("X and Y uncorrelated. Cannot calculate lenght & width. Exiting ...")
```

Comment:
this should raise an exception, not print and exit. Ideally,  define a custom exception like `HillasParameterizationError` that is a subclass of `RuntimeError`  and raise it here, with the description as the same text you would have printed.